### ISSUE TITLE
[RFC] Streaming request & responses

### ISSUE THREAD

**Issue by lwc** [spiral/roadrunner-plugins#29]
This project looks like it could be very useful in building a GRPC server bridge for PHP, which would be amazing.

GRPC has the concept of streaming request & responses, is that something you'd consider adding to roadrunner/goridge?

Thanks

**Comment by rustatian** [comment]
A little update about streaming file responses. The `sendfile` HTTP middleware handles the `X-Sendfile` header and streams the response back w/o involving the worker in the process. So, you are able to handle the requests with the PHP worker and stream the file(s) in parallel. More info here: https://github.com/spiral/roadrunner-plugins/issues/9, PR https://github.com/spiral/roadrunner-plugins/pull/56.

**Comment by Ocramius** [comment]
@rustatian streaming files usually not the problem though: streaming batches (think "massive CSV generated from DB") is still the issue though

**Comment by rustatian** [comment]
> @rustatian streaming files usually not the problem though: streaming batches (think "massive CSV generated from DB") is still the issue though

Yep, sure. That's was just a little update about streaming in general. Not about the ticket's topic.
